wayland-display-core/src/comp/focus.rs

```diff
@@ -117,21 +120,84 @@ impl PointerTarget<super::State> for FocusTarget {
         }
     }
 
-    fn button(&self, seat: &Seat<super::State>, data: &mut super::State, event: &ButtonEvent) {
+    fn button(&self, seat: &Seat<State>, data: &mut State, event: &ButtonEvent) {
         match self {
             FocusTarget::Wayland(w) => w.button(seat, data, event),
             FocusTarget::Popup(p) => p.wl_surface().button(seat, data, event),
         }
     }
 
-    fn axis(&self, seat: &Seat<super::State>, data: &mut super::State, frame: AxisFrame) {
+    fn axis(&self, seat: &Seat<State>, data: &mut State, frame: AxisFrame) {
         match self {
             FocusTarget::Wayland(w) => w.axis(seat, data, frame),
             FocusTarget::Popup(p) => p.wl_surface().axis(seat, data, frame),
         }
     }
 
-    fn leave(&self, seat: &Seat<super::State>, data: &mut super::State, serial: Serial, time: u32) {
+    fn frame(&self, seat: &Seat<State>, data: &mut State) {
+        match self {
+            FocusTarget::Wayland(w) => w.frame(seat, data),
+            FocusTarget::Popup(p) => { todo!("the trait `PointerTarget<_>` is not implemented for `&WlSurface`") }
```

Comment:
This is still not working..

Reply:
Well there certainly is an implementation for it: https://smithay.github.io/smithay/smithay/input/pointer/trait.PointerTarget.html#impl-PointerTarget%3CD%3E-for-WlSurface

How exactly was your attempt looking? Something like this?
```suggestion
            FocusTarget::Popup(p) => { PointerTarget::frame(p.wl_surface(), seat, data) }
```